File: IntegrationTests/Parser/TypeParsing/FragmentUsageTests/ClassWithMultipleFragments.cs
using System;
using InjectionCop.Fragment;

namespace InjectionCop.IntegrationTests.Parser.TypeParsing.FragmentUsageTests
{
  public class ClassWithMultipleFragments
  {
    public void InvalidFragmentUsage (
        [Fragment ("FirstFragment")] [SqlFragment ()] int parameter)
    {
    }

    public void ValidFragmentUsage (
        [Fragment ("FirstFragment")] int parameter)
    {
    }

    public void NoFragmentUsage (
        int parameter)
    {
    }
  }
}

File: IntegrationTests.VB/Parser/TypeParsing/FragmentUsageTests/ClassWithMultipleFragments.vb
Imports InjectionCop.Fragment
Imports System

Namespace Parser.TypeParsing.FragmentUsageTests
	Public Class ClassWithMultipleFragments
		Public Sub InvalidFragmentUsage(<Fragment("FirstFragment"), SqlFragment()> parameter As Integer)
		End Sub

		Public Sub ValidFragmentUsage(<Fragment("FirstFragment")> parameter As Integer)
		End Sub

		Public Sub NoFragmentUsage(parameter As Integer)
		End Sub
	End Class
End Namespace
